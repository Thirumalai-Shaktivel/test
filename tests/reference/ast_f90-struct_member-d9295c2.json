{
    "basename": "ast_f90-struct_member-d9295c2",
    "cmd": "lfortran --show-ast-f90 --no-color {infile}",
    "infile": "tests/struct_member.f90",
    "infile_hash": "d5f350c491fea9d256524365662119c9ffec6095ad01693043261210",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_f90-struct_member-d9295c2.stdout",
    "stdout_hash": "9cf01bda706c54604bfa1d1db201ef11495db463bd0aed444958254b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}