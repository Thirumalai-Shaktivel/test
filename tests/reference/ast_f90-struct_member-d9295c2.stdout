pure function name(self)
class, intent(in) :: self
logical :: finished
integer :: ii
finished = all(self%dep(:self%ndep)%done)
end function name

pure function find_name(self, name) result(pos)
class, intent(in) :: self
character(len=*), intent(in) :: name
integer :: pos
integer :: ii
pos = 0
do ii = 1, self%ndep
    if ((name)==(self%dep(ii)%name)) then
        pos = ii
        exit
    end if
end do
end function find_name
